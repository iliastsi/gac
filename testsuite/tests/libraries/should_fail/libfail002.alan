input():proc
  i : int;
  b : byte;
  s : byte[42];
{
  b = readInteger();
  s = readInteger();

  i = readByte();
  s = readByte();

  i = readChar();
  s = readChar();

  readString(b, s);
  readString(s, s);
  readString(i, i);
  readString(i, b);
  readString(i);
}
