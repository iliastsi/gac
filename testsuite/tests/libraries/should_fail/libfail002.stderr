libfail002.alan:6:3: Error: Type mismatch at `b = readInteger();'
	Lvalue is of type `byte' but Rvalue is of type `int'

libfail002.alan:7:3: Error: Type mismatch at `s = readInteger();'
	Lvalue is of type `array of byte' but Rvalue is of type `int'

libfail002.alan:9:3: Error: Type mismatch at `i = readByte();'
	Lvalue is of type `int' but Rvalue is of type `byte'

libfail002.alan:10:3: Error: Type mismatch at `s = readByte();'
	Lvalue is of type `array of byte' but Rvalue is of type `byte'

libfail002.alan:12:3: Error: Type mismatch at `i = readChar();'
	Lvalue is of type `int' but Rvalue is of type `byte'

libfail002.alan:13:3: Error: Type mismatch at `s = readChar();'
	Lvalue is of type `array of byte' but Rvalue is of type `byte'

libfail002.alan:15:14: Error: Type mismatch at `b'
	Incompatible type of argument 1 of function `readString'
	Expected `int' but argument is of type `byte'

libfail002.alan:16:14: Error: Type mismatch at `s'
	Incompatible type of argument 1 of function `readString'
	Expected `int' but argument is of type `array of byte'

libfail002.alan:17:17: Error: Type mismatch at `i'
	Incompatible type of argument 2 of function `readString'
	Expected `array of byte' but argument is of type `int'

libfail002.alan:18:17: Error: Type mismatch at `b'
	Incompatible type of argument 2 of function `readString'
	Expected `array of byte' but argument is of type `byte'

libfail002.alan:19:3: Error: Type mismatch at `readString(i)'
	The function `readString' is applied to 1 parameters but its type has 2

